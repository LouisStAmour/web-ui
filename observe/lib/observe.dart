// Copyright (c) 2013, the Dart project authors.  Please see the AUTHORS file
// for details. All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.

/**
 * *Warning*: this library is experimental, and APIs are subject to change.
 *
 * This library is used to observe changes to [Observable] types. It also
 * has helpers to implement [Observable] objects.
 *
 * For example:
 *
 *     class Monster extends ObservableMixin {
 *       int _health = 100;
 *       static const _HEALTH = 'health';
 *       get health => _health;
 *       set health(value) {
 *         _health = notifyChange(_HEALTH, _health, value);
 *       }
 *       void damage(int amount) {
 *         print('$this takes $amount damage!');
 *         health -= amount;
 *       }
 *       toString() => 'Monster';
 *
 *       // These methods are temporary until dart2js supports mirrors.
 *       getValue(key) {
 *         if (key == _HEALTH) return health;
 *         return null;
 *       }
 *       setValue(key, val) {
 *         if (key == _HEALTH) health = val;
 *       }
 *     }
 *
 *     main() {
 *       var obj = new Monster();
 *       obj.changes.listen((records) {
 *         print('Changes to $obj were: $records');
 *       });
 *       // Asynchronously schedules delivery of these changes
 *       obj.damage(10);
 *       obj.damage(20);
 *       print('done!');
 *     }
 */
library observe;

import 'dart:async';
import 'dart:collection';
import 'dart:math' as math;

// TODO(jmesserly): use libraries instead of parts.
part 'src/list_diff.dart';
part 'src/observe_path.dart';
part 'src/observable.dart';
part 'src/observable_box.dart';
part 'src/observable_list.dart';
part 'src/observable_map.dart';
