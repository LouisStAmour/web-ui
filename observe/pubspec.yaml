name: observe
author: "Web UI Authors <web-ui-dev@dartlang.org>"
homepage: https://github.com/dart-lang/web-ui
description: observable objects for data binding
version: 0.0.1
dependencies:
  unittest: ">=0.4.6"
environment:
  sdk: ">=0.4.6"
