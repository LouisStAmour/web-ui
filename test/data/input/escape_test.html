<!DOCTYPE html>
<!--
Copyright (c) 2012, the Dart project authors.  Please see the AUTHORS file
for details. All rights reserved. Use of this source code is governed by a
BSD-style license that can be found in the LICENSE file.
-->
<html lang="en">
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
  <script src="packages/web_ui/testing/testing.js"></script>
</head>
<body>
  <!-- Tests that you can inject HTML using SafeHtml. -->
  <div>binding as text (expect &lt;span&gt;hello&lt;/span&gt;): {{text}}</div>
  <div>binding as SafeHtml (expect hello inside a real span): {{html}}</div>
  <script type="application/dart">
  import 'dart:html';
  import 'package:unittest/unittest.dart';
  import 'package:web_ui/observe.dart';
  import 'package:web_ui/safe_html.dart';

  @observable var text = '<span>hello</span>';
  @observable var html = new SafeHtml.unsafe('<span>hello</span>');
  main() {
    window.postMessage('done', '*');
  }
  </script>
</body>
</html>
