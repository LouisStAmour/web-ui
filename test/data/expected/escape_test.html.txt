Content-Type: text/plain
<html lang="en"><head><style>template { display: none; }</style>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
  <script src="../input/packages/web_ui/testing/testing.js"></script>
</head>
<body>
  <!-- Tests that you can inject HTML using SafeHtml. -->
  <div>binding as text (expect &lt;span&gt;hello&lt;/span&gt;): &lt;span&gt;hello&lt;/span&gt;</div>
  <div>binding as SafeHtml (expect hello inside a real span): <span>hello</span></div>
  


<script type="text/javascript" src="packages/web_ui/shadowdom.debug.js"></script>
<script type="application/dart" src="escape_test.html_bootstrap.dart"></script></body></html>
